diff --git a/contribs/Makefile.am b/contribs/Makefile.am
index 3f6fac0a85..6c466b4eb5 100644
--- a/contribs/Makefile.am
+++ b/contribs/Makefile.am
@@ -1 +1 @@
-SUBDIRS = cray lua mic nss_slurm openlava pam pam_slurm_adopt perlapi pmi pmi2 seff sgather sgi sjobexit torque
+SUBDIRS = lua mic nss_slurm openlava pam pam_slurm_adopt perlapi pmi pmi2 seff sgather sgi sjobexit torque
diff --git a/src/plugins/burst_buffer/Makefile.am b/src/plugins/burst_buffer/Makefile.am
index c187bb83bd..aabd7f3fe8 100644
--- a/src/plugins/burst_buffer/Makefile.am
+++ b/src/plugins/burst_buffer/Makefile.am
@@ -2,6 +2,3 @@
 
 SUBDIRS = generic
 
-if WITH_JSON_PARSER
-SUBDIRS += datawarp
-endif
diff --git a/src/plugins/core_spec/Makefile.am b/src/plugins/core_spec/Makefile.am
index 44b1ed67c0..f06bc541d7 100644
--- a/src/plugins/core_spec/Makefile.am
+++ b/src/plugins/core_spec/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile for core specialization plugins
 
-SUBDIRS = cray_aries none
+SUBDIRS = none
diff --git a/src/plugins/job_submit/Makefile.am b/src/plugins/job_submit/Makefile.am
index e393bcfb1c..b98cf2e000 100644
--- a/src/plugins/job_submit/Makefile.am
+++ b/src/plugins/job_submit/Makefile.am
@@ -2,7 +2,6 @@
 
 SUBDIRS = \
 	all_partitions \
-	cray_aries \
 	defaults \
 	logging \
 	lua \
diff --git a/src/plugins/mpi/Makefile.am b/src/plugins/mpi/Makefile.am
index 4d055e2030..1a44dfb2ab 100644
--- a/src/plugins/mpi/Makefile.am
+++ b/src/plugins/mpi/Makefile.am
@@ -1,6 +1,6 @@
 # Makefile for mpi plugins
 
-SUBDIRS = cray_shasta none pmi2
+SUBDIRS = none pmi2
 if HAVE_PMIX
 SUBDIRS += pmix
 endif
diff --git a/src/plugins/node_features/Makefile.am b/src/plugins/node_features/Makefile.am
index f5ef036303..d955c51ba1 100644
--- a/src/plugins/node_features/Makefile.am
+++ b/src/plugins/node_features/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile for node_features plugins
 
-SUBDIRS = knl_cray knl_generic
+SUBDIRS = knl_generic
diff --git a/src/plugins/power/Makefile.am b/src/plugins/power/Makefile.am
index ccde44c322..c6171740c1 100644
--- a/src/plugins/power/Makefile.am
+++ b/src/plugins/power/Makefile.am
@@ -1,3 +1,3 @@
 # Makefile for power management plugins
 
-SUBDIRS = common cray_aries none
+SUBDIRS = common none
diff --git a/src/plugins/select/Makefile.am b/src/plugins/select/Makefile.am
index 38bae0ff5c..ca7133f43f 100644
--- a/src/plugins/select/Makefile.am
+++ b/src/plugins/select/Makefile.am
@@ -1,7 +1,7 @@
 # Makefile for node selection plugins
 
 # NOTE: other must be in the list before cray_aries lest the build break
-SUBDIRS = other cons_common cons_res cons_tres cray_aries linear
+SUBDIRS = other cons_common cons_res cons_tres linear
 
 # Each plugin here needs a plugin_id.
 # Here are the current plugin_ids for each plugin.
diff --git a/src/plugins/switch/Makefile.am b/src/plugins/switch/Makefile.am
index 365f5b8d93..dc56a01a55 100644
--- a/src/plugins/switch/Makefile.am
+++ b/src/plugins/switch/Makefile.am
@@ -1,7 +1,3 @@
 # Makefile for switch plugins
 
 SUBDIRS = none
-
-if WITH_SWITCH_CRAY_ARIES
-SUBDIRS += cray_aries
-endif
diff --git a/src/plugins/task/Makefile.am b/src/plugins/task/Makefile.am
index f79282f3e6..356cdcd500 100644
--- a/src/plugins/task/Makefile.am
+++ b/src/plugins/task/Makefile.am
@@ -1,6 +1,6 @@
 # Makefile for task plugins
 
-SUBDIRS = affinity cray_aries none
+SUBDIRS = affinity none
 
 if WITH_CGROUP
 SUBDIRS += cgroup
