diff --git a/contribs/cray/csm/slurmconfgen_smw.py b/contribs/cray/csm/slurmconfgen_smw.py
index 1d1aadedaf..b2f65c242b 100644
--- a/contribs/cray/csm/slurmconfgen_smw.py
+++ b/contribs/cray/csm/slurmconfgen_smw.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 #
 # Copyright 2015-2016 Hewlett Packard Enterprise Development LP
 """ A script to generate slurm.conf and gres.conf for a
diff --git a/contribs/cray/slurmconfgen.py.in b/contribs/cray/slurmconfgen.py.in
index 31de1a0c28..269b1d55a7 100644
--- a/contribs/cray/slurmconfgen.py.in
+++ b/contribs/cray/slurmconfgen.py.in
@@ -1,4 +1,4 @@
-#!/usr/bin/python
+#!/usr/bin/python3
 #
 # Copyright 2013 Hewlett Packard Enterprise Development LP
 #
